src/sys/unix/tcp.rs

```diff
@@ -50,3 +50,52 @@ pub fn bind(addr: SocketAddr) -> io::Result<net::TcpListener> {
         .map(|_| unsafe { net::TcpListener::from_raw_fd(socket) })
     })
 }
+
+pub fn accept(listener: &net::TcpListener) -> io::Result<(net::TcpStream, SocketAddr)> {
+    let mut addr: MaybeUninit<libc::sockaddr_storage> = MaybeUninit::uninit();
+    let mut length = size_of::<libc::sockaddr_storage>() as libc::socklen_t;
+
+    // On platforms that support it we can use `accept4(2)` to set `NONBLOCK`
+    // and `CLOEXEC` in the call to accept the connection.
+    #[cfg(any(
+        target_os = "android",
+        target_os = "dragonfly",
+        target_os = "freebsd",
+        target_os = "linux",
+        target_os = "openbsd"
+    ))]
+    let stream = {
+        syscall!(accept4(
+            listener.as_raw_fd(),
+            addr.as_mut_ptr() as *mut _,
+            &mut length,
+            libc::SOCK_CLOEXEC | libc::SOCK_NONBLOCK,
+        ))
+        .map(|socket| unsafe { net::TcpStream::from_raw_fd(socket) })
+    }?;
+
+    // But not all platforms have the `accept4(2)` call. Luckily BSD (derived)
+    // OSes inherit the non-blocking flag from the listener, so we just have to
+    // set `CLOEXEC`.
```

Comment:
> When using  accept(), portable programs should not rely on the O_NONBLOCK
> and O_ASYNC properties and  the signal destination being inherited, but
> should set them explicitly using fcntl(2); accept4() sets these properties
> consistently, but may not be fully portable across UNIX platforms.

This makes sense. The "portable" programs are eiter going to have `accept4`, or we can assume inherit `NONBLOCK`.

Reply:
I've manually checked all the OSes and the list and they all inherit the `NONBLOCK` property, it seems only Linux really diverges from this.